### ISSUE TITLE
Target directory in output folder

### ISSUE THREAD

**Issue by 0x1bi** [richfelker/musl-cross-make#112]
Is it possible to rather than the header files, libraries, and linker to be placed rather than into the target directory in sysroot to the root output directory.

So from this...

```
output
    |
    + - bin
    + - lib
    + - ...
    + - x86_64-linux-musl
        + - bin
        + - lib
        + - ...
```

To this...

```
output
    |
    + - bin
    + - lib
    + - ...
```

Thank you in advance.

**Comment by richfelker** [comment]
You would use `--with-sysroot` not `--with-build-sysroot` for this, but I don't have any idea if it will work - mcm is not setup to be installed like that, and it's rather going against logical organization, since the top-level output dir is binaries/libraries/etc. for the *host* (the system you run the cross toolchain on) and the `$target`-named directory under that is binaries/libraries/etc. for the *target* (the system you're cross-compiling for).

**Comment by 0x1bi** [comment]
I attempted to compile with `--with-sysroot` however to no avail; the target directory was still present in the output directory. Is there any other option which can possibly build the toolchain in the way that I've described?

Also, why is there no compiler in the target directory, e.g. `output/x86_64-linux-musl/bin/cc`?

**Comment by richfelker** [comment]
> I attempted to compile with `--with-sysroot` however to no avail; the target directory was still present in the output directory. Is there any other option which can possibly build the toolchain in the way that I've described?

You'd need to hack this into the makefile since, as written, it sets the sysroot itself. But I want to reiterate that this is not a good idea. It's gratuitously breaking the intended split between host and target files and unless I'm missing something you don't have any technical reason to do this, just an aesthetic one that seems to be based on misunderstanding the purpose of the existing layout.

> Also, why is there no compiler in the target directory, e.g. `output/x86_64-linux-musl/bin/cc`?

Because building a cross compiler and cross-compiling a native compiler for the target are two completely different tasks, and doing them both when you don't need the latter would double the time and space required to build.

**Comment by 0x1bi** [comment]
>Because building a cross compiler and cross-compiling a native compiler for the target are two completely different tasks, and doing them both when you don't need the latter would double the time and space required to build.

That's fair. Is there not a `NATIVE` option I can set in config.mak?

**Comment by richfelker** [comment]
Yes, mcm can also cross-compile native compilers for your target, and in this case they don't have the cross directory structure you're unhappy with. You need to already have the cross compiler in your PATH, then run mcm with NATIVE=y, and it should just work.

**Comment by 0x1bi** [comment]
I set the compuler to my path and set `NATIVE=1` in `COMMON_CONFIG` however my output directory still has the x86_64-linux-musl directory. Is this the intended result for building a native compiler? When looking into the bin dir in the x86_64... dir I can't seem to find the compiler.

**Comment by 0x1bi** [comment]
Upon adding `NATIVE=y` to `config.mak` I receive the following error:

```
make[1]: Entering directory '/home/ben/musl-cross-make/build/x86_64-linux-musl/x86_64-linux-musl'
mkdir -p obj_musl
ln -sf ../../../musl-1.2.1 src_musl
cd obj_musl && ../src_musl/configure  --prefix= --host=x86_64-linux-musl
checking for C compiler...
../src_musl/configure: cannot find a C compiler
make[1]: *** [Makefile:226: obj_musl/.lc_configured] Error 1
make[1]: Leaving directory '/home/ben/musl-cross-make/build/x86_64-linux-musl/x86_64-linux-musl'
make: *** [Makefile:182: all] Error 2
```

I've checked to make sure that the compiler exists in the output directory

**Comment by 0x1bi** [comment]
After some playing around with the environment I was able to compile a native Musl GCC compiler. Thanks @richfelker !

**Comment by clark800** [comment]
It would be really helpful to have a section in the readme on how to build a native compiler. Also it would be nice if it was scripted so it could be built in a single run of `make`.

**Comment by SomethingGeneric** [comment]
would anyone be willing to share an implementation like our now-departed friend who started this thread>
